src/Json.Schema.ToDotNet.UnitTests/DataModelGeneratorTests.cs

```diff
@@ -294,6 +294,191 @@ public int Compare(C left, C right)
+        [DataMember(Name = ""integerProperty_min_0_max_na"", IsRequired = false, EmitDefaultValue = false)]
+        public int IntegerProperty_min_0_max_na { get; set; }
+        [DataMember(Name = ""integerProperty_min_minus1_max_na"", IsRequired = false, EmitDefaultValue = false)]
+        public int IntegerProperty_min_minus1_max_na { get; set; }
+        [DataMember(Name = ""integerProperty_min_1_max_na"", IsRequired = false, EmitDefaultValue = false)]
+        public int IntegerProperty_min_1_max_na { get; set; }
+        [DataMember(Name = ""integerProperty_min_na_max_0"", IsRequired = false, EmitDefaultValue = false)]
+        public int IntegerProperty_min_na_max_0 { get; set; }
+        [DataMember(Name = ""integerProperty_min_0_max_int64times10"", IsRequired = false, EmitDefaultValue = false)]
+        public int IntegerProperty_min_0_max_int64times10 { get; set; }
+        [DataMember(Name = ""integerProperty_min_0_max_int64"", IsRequired = false, EmitDefaultValue = false)]
+        public int IntegerProperty_min_0_max_int64 { get; set; }
+        [DataMember(Name = ""integerProperty_min_0_max_int32add1"", IsRequired = false, EmitDefaultValue = false)]
+        public int IntegerProperty_min_0_max_int32add1 { get; set; }
+        [DataMember(Name = ""integerProperty_min_0_max_int32"", IsRequired = false, EmitDefaultValue = false)]
+        public int IntegerProperty_min_0_max_int32 { get; set; }
+        [DataMember(Name = ""integerProperty_min_int32_max_0"", IsRequired = false, EmitDefaultValue = false)]
+        public int IntegerProperty_min_int32_max_0 { get; set; }
+        [DataMember(Name = ""integerProperty_min_int32minus1_max_0"", IsRequired = false, EmitDefaultValue = false)]
+        public int IntegerProperty_min_int32minus1_max_0 { get; set; }
+        [DataMember(Name = ""integerProperty_min_int64_max_0"", IsRequired = false, EmitDefaultValue = false)]
+        public int IntegerProperty_min_int64_max_0 { get; set; }
+        [DataMember(Name = ""integerProperty_min_int64times10_max_0"", IsRequired = false, EmitDefaultValue = false)]
+        public int IntegerProperty_min_int64times10_max_0 { get; set; }
+        [DataMember(Name = ""integerProperty_min_huge_max_0"", IsRequired = false, EmitDefaultValue = false)]
+        public int IntegerProperty_min_huge_max_0 { get; set; }
+        [DataMember(Name = ""integerProperty_min_0_max_huge"", IsRequired = false, EmitDefaultValue = false)]
+        public int IntegerProperty_min_0_max_huge { get; set; }
+    }
+}";
+            GeneratesPropertiesWithIntegerTypes_Helper(GenerateIntegerOption.Auto, ExpectedClass_Auto);
+            GeneratesPropertiesWithIntegerTypes_Helper(GenerateIntegerOption.BigInteger, ExpectedClass_BigInteger);
+            GeneratesPropertiesWithIntegerTypes_Helper(GenerateIntegerOption.Long, ExpectedClass_Long);
+            GeneratesPropertiesWithIntegerTypes_Helper(GenerateIntegerOption.Int, ExpectedClass_Int);
```

Comment:
These all seem like happy path test cases. Can we explicitly have a test case where the user has set `--generate-integer-as int` but the actual value will cause an int overflow?

Reply:
Additional test case needed: `--generate-integer-as` is not set and default applies. Both happy path and with an int overflow.

Reply:
1. the input is a .json schema, and output is C# classes files, there is no actual value involved, the goal is user specify to generate int in his C# class and we output int.
2. added the test that when user does not include the parameter, we will be getting int.